### PR TITLE
added "sections" and layout improvments and sign up and login bug fixes

### LINKED ISSUE #176: Create Episode Sections component
## **Description:** 
- **Brief Description:** Navigational tool for episode sections.
- **Purpose/Functionality:** Divides episodes into sections for easy navigation.
- **Related Page -frontend:** NowPlaying page.
- **Related Files/Modules -backend:** Episode segmentation module.
- **Additional Notes:** Sections should be clearly labeled for user convenience.
## Acceptance Criteria:  
- [x] Clear and labeled sections within the episode.
- [x] Seamless navigation between sections.

### PR DESCRIPTION
**Pull request by Radwanbaba13**
This pull request adds a series of commits that have fully implemented the sections feature on the frontend, including:

-  A component to view all sections, where you can manage existing sections such as deleting them.
- The ability to add sections, by adding a title and timestamps which are set with the help of a player bar, to facilitate the creation process for the user.  
- updated layouts and folder structures. 
- fixed  sign up bugs 

Closes #176 #235 #246 #238

### CODE REVIEW

**Review by mathieutrudeau — CHANGES_REQUESTED** [review]
Please add the cypress screenshots and vids to the gitignore.

**Review by mathieutrudeau — APPROVED** [review]
Looks good.

**Review by jonathan-githubofficial — CHANGES_REQUESTED** [review]
don't approve just yet

File: frontend/pages/NowPlaying/[EpisodeId].tsx
Comment (jonathan-githubofficial) [inline frontend/pages/NowPlaying/[EpisodeId].tsx:1]: the section's useEffect should be handled in the "Sections" component. the section component in NowPlaying should only accept episode ID .

File: frontend/components/nowPlaying/Sections.tsx
Comment (jonathan-githubofficial) [inline frontend/components/nowPlaying/Sections.tsx:1]: the section's useEffect should be handled in the "Sections" component. the section component in NowPlaying should only accept episode ID .

**Review by jonathan-githubofficial — CHANGES_REQUESTED** [review]

**Review by jonathan-githubofficial — APPROVED** [review]

**Pull request merged**